packages/discovery-provider/src/challenges/challenge_event_bus.py

```diff
@@ -101,18 +104,28 @@ def dispatch(
         # Sanitize input, drop the event if it's malformed
         valid_event = event is not None and isinstance(event, str)
         valid_block = block_number is not None and isinstance(block_number, int)
+        valid_block_datetime = block_datetime is not None and isinstance(
```

Comment:
this seems a bit scarier than before, wonder if there's any chance we'd get a malformed `block_datetime`.. probably fine tho

Reply:
we're doing thorough QA for all challenges so we should know!